app/main/lib/reindex_images.py

```diff
@@ -0,0 +1,46 @@
+from flask import current_app as app
+from flask import request, current_app as app
+from flask import current_app as app
+from app.main import db
+from app.main.model.image import ImageModel
+from app.main.lib.helpers import context_matches
+from app.main.lib.similarity_helpers import get_context_query, drop_context_from_record
+from app.main.lib.image_similarity import add_image_pdq
+
+from sqlalchemy import text
+from sqlalchemy.orm.attributes import flag_modified
+from sqlalchemy.orm.exc import NoResultFound
+
+app.app_context()
+def _after_log(retry_state):
+  app.logger.debug("Retrying image similarity...")
+
+def get_all_images():
+  try:
+    cmd = """
+      SELECT id, sha256, url, context,created_at,pdq  FROM images
+    """
+    matches = db.session.execute(text(cmd)).fetchall()
+    keys = ('id', 'sha256', 'url', 'context','created_at','pdq')
+    rows = []
+    for values in matches:
+      row = dict(zip(keys, values))
+      if values['pdq'] is not None:
```

Comment:
This logic is backwards. We want rows where pdq IS NULL (this is the opposite). As noted above, however, the filter should be in the SQL query, which means we won't need this line.